jsonschema/exceptions.py

```diff
@@ -23,6 +24,8 @@
 WEAK_MATCHES: frozenset[str] = frozenset(["anyOf", "oneOf"])
 STRONG_MATCHES: frozenset[str] = frozenset()
 
+JSON_PATH_COMPATIBLE_PROPERTY_PATTERN = re.compile("^[a-zA-Z][a-zA-Z0-9_]*$")
```

Comment:
Please make sure this is private even if we don't use `jsonpath-ng` in the real code yet. Though if it turns out that a regex isn't good enough for fully accurate generation of paths then I think we should think carefully about adding the dep even now perhaps.

Reply:
This is addressed.